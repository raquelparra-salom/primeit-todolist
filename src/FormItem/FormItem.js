import * as React from 'react';
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogTitle from '@mui/material/DialogTitle';

function FormItem({ open, item, handleClickClose, handleClickAdd, handleChangeItem }) {


    return (
        <div>
            <Dialog open={open} onClose={handleClickClose}>
                <DialogTitle>Add item to list</DialogTitle>
                <DialogContent>
                    <TextField
                        onChange={handleChangeItem}
                        value={item}
                        autoFocus
                        margin="dense"
                        id="name"
                        label="Type the text here"
                        type="text"
                        fullWidth
                        variant="standard"
                    />
                </DialogContent>
                <DialogActions>
                    <Button disabled={item === ""} variant="contained" onClick={handleClickAdd}>Add</Button>
                    <Button onClick={handleClickClose}>Cancel</Button>
                </DialogActions>
            </Dialog>
        </div>
    );
}

export default FormItem;