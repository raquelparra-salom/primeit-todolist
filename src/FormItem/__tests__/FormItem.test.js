import React from 'react';
import { render } from '@testing-library/react';
import FormItem from '../FormItem';

test('render formItem close', () => {

    const handleClickClose = () => { }
    const handleClickAdd = () => { }
    const handleChangeItem = () => { }
    const { container } = render(
        <FormItem
            open={false}
            item="Texto ejemplo"
            handleClickClose={handleClickClose}
            handleClickAdd={handleClickAdd}
            handleChangeItem={handleChangeItem}
        />);
    expect(container).toMatchSnapshot();
});

test('render formItem open', () => {

    const handleClickClose = () => { }
    const handleClickAdd = () => { }
    const handleChangeItem = () => { }
    const { container } = render(
        <FormItem
            open={true}
            item="Texto ejemplo"
            handleClickClose={handleClickClose}
            handleClickAdd={handleClickAdd}
            handleChangeItem={handleChangeItem}
        />);
    expect(container).toMatchSnapshot();
});
