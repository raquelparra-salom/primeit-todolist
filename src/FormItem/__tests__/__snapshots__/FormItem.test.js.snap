// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render formItem close 1`] = `
<div>
  <div />
</div>
`;

exports[`render formItem open 1`] = `
<div
  aria-hidden="true"
>
  <div />
</div>
`;
